### PR TITLE
provide wb_dims()

### PR DESCRIPTION
**Pull request by JanMarvin**
@olivroy something I quickly typed down in addition to #681, thinking about `ggplot2::aes()` and how clean this approach is. This PR provides `wb_dims()` which can take either `row`/`col`, `rows/cols` or objects that can be coerced via `as.data.frame()`.

``` r
  library(openxlsx2)
  
  # take data frames and col_names & row_names
  wb_dims(mtcars)
  wb_dims(mtcars, col_names = TRUE, row_names = TRUE)
  
  # take vectors
  wb_dims(letters)
  wb_dims(t(letters))
  
  # or time series
  wb_dims(AirPassengers)
  
  # single values
  wb_dims(1)
  
  # named 
  wb_dims(rows =  1:10, cols = 5:7)
  # wb_dims(rows =  1:10, row = 5:7)
  
  # single cell
  wb_dims(row = 5, col = 7)
  
  # two vectors  
  wb_dims(1:10, LETTERS)
  wb_dims(1:10, 1:26)
  
  # in a wb chain
  wb <- wb_workbook()$
    add_worksheet()$
    add_data(x = mtcars)$
    add_fill(
      dims = wb_dims(mtcars, col_names = TRUE),
      color = wb_color("yellow")
    )
```

~~This could be tweaked further, but might provide a simpler helper for users. A name to indicate that it is coming from `openxlsx` could be `wb_dims()`.~~

Please let me know what you think!

[Edit:] Updated

### CODE REVIEW

**Comment by JanMarvin** [comment]
`wb_dims()` might be better, because `dims` is just an `s` away from `dim` and a small typo might quickly cause confusion.

**Comment by olivroy** [comment]
Oh yes, I like that, especially with the prefix! 

With this, there wouldn't be a use-case for rowcol_to_dims() in the exported API? 

could inform users of the new syntax. 

But I think it's a step forward for a consistent, streamlined API

**Comment by JanMarvin** [comment]
Yes, this would be a more convenient and possibly simpler to remember function that creates the same output. It's a wrapper around functions.

File: R/utils.R [hunk 1]
```diff
@@ -259,6 +259,70 @@ rowcol_to_dim <- function(row, col) {
...
+  args <- list(...)
+  nams <- names(args)
+
+  col_names <- args$col_names
+  row_names <- args$row_names
+
+  has_cnam <- is.null(col_names)
+  has_rnam <- is.null(row_names)
+
+  if (has_cnam) col_names <- FALSE
```
Comment (JanMarvin) [inline R/utils.R:282]: default to true

**Comment by olivroy** [comment]
Hi @JanMarvin ,

after reviewing, `wb_dims()` may be enhanced 

* provide data_only option maybe?
 * For example in a vignette 
```r
wb_dims(mtcars)
"A1:K33"
# I propose:
wb_dims(mtcars, data_only = TRUE)
"A2:K33"
# it could be very useful when applying styling to the data.
```
* Error if only `cols` or `rows` is provided. 

```r
# now
wb_dims(cols = 4)
"A1"
# I suggest 
wb_dims(cols = 4)
"D1"
```

I can file a new issue if it's easier.

**Comment by JanMarvin** [comment]
> wb_dims(x = mtcars, cols = "vs")

like `wb_dims(mtcars, start_cols = "VS")`?

But yes, I though about the same (regarding columns, rownames and data selection). Go for it if you want to improve it. I usually use the cell directly. Open a new issue or a pull request, I currently don't have time for this.

**Pull request merged**